web-local/src/lib/http-request-queue/HttpRequestQueue.ts

```diff
@@ -131,6 +134,8 @@ export class HttpRequestQueue {
   }
 
   private async popEntries() {
+    appQueryStatusStore.set(!!this.nameHeap.peek()?.name);
```

Comment:
This should be based on `activeCount`. When we pop from heap we just trigger the network call. So the spinner should be spinning when `activeCount` is not 0.